lms/djangoapps/instructor/views/api.py

```diff
@@ -2686,27 +2687,23 @@ def remove_certificate_exception(course_key, student):
     :param student: User object whose certificate exception needs to be removed.
     :return:
     """
-    try:
-        certificate_exception = certs_api.get_allowlist_entry(student, course_key)
-    except ObjectDoesNotExist:
+    log.info(f"Request received to remove an allowlist entry for student {student.id} in course {course_key}.")
+
+    allowlist_entry = certs_api.get_allowlist_entry(student, course_key)
+    if not allowlist_entry:
         raise ValueError(  # lint-amnesty, pylint: disable=raise-missing-from
             _('Certificate exception (user={user}) does not exist in certificate white list. '
               'Please refresh the page and try again.').format(user=student.username)
         )
 
-    try:
```

Comment:
The `get_certificate_for_user` function was already catching the ObjectDoesNotExist exception and returning None, this was a change in previous behavior (where we were using the model directly to access/look up the certificate for the user). I missed this when I made the change.